Lib/fontTools/varLib/instancer.py

```diff
@@ -158,24 +159,71 @@ def setMvarDeltas(varfont, deltas):
             )
 
 
-def instantiateMvar(varfont, location):
+def instantiateMVAR(varfont, location):
     log.info("Instantiating MVAR table")
 
     mvar = varfont["MVAR"].table
     fvarAxes = varfont["fvar"].axes
-    defaultDeltas, varIndexMapping = instantiateItemVariationStore(
-        mvar.VarStore, fvarAxes, location
-    )
+    varStore = mvar.VarStore
+    defaultDeltas = instantiateItemVariationStore(varStore, fvarAxes, location)
     setMvarDeltas(varfont, defaultDeltas)
 
-    if varIndexMapping:
+    if varStore.VarRegionList.Region:
+        varIndexMapping = varStore.optimize()
         for rec in mvar.ValueRecord:
             rec.VarIdx = varIndexMapping[rec.VarIdx]
     else:
-        # Delete table if no more regions left.
         del varfont["MVAR"]
 
 
+def _remapVarIdxMap(table, attrName, varIndexMapping, glyphOrder):
+    oldMapping = getattr(table, attrName).mapping
+    newMapping = [varIndexMapping[oldMapping[glyphName]] for glyphName in glyphOrder]
+    setattr(table, attrName, builder.buildVarIdxMap(newMapping, glyphOrder))
+
+
+# TODO(anthrotype) Add support for HVAR/VVAR in CFF2
+def _instantiateVHVAR(varfont, location, tableFields):
+    tableTag = tableFields.tableTag
+    fvarAxes = varfont["fvar"].axes
+    # Deltas from gvar table have already been applied to the hmtx/vmtx. For full
+    # instances (i.e. all axes pinned), we can simply drop HVAR/VVAR and return
+    if set(location).issuperset(axis.axisTag for axis in fvarAxes):
+        log.info("Dropping %s table", tableTag)
+        del varfont[tableTag]
+        return
+
+    log.info("Instantiating %s table", tableTag)
+    vhvar = varfont[tableTag].table
+    varStore = vhvar.VarStore
+    # since deltas were already applied, the return value here is ignored
+    instantiateItemVariationStore(varStore, fvarAxes, location)
+
+    if varStore.VarRegionList.Region:
+        if getattr(vhvar, tableFields.advMapping):
+            varIndexMapping = varStore.optimize()
+            glyphOrder = varfont.getGlyphOrder()
+            _remapVarIdxMap(vhvar, tableFields.advMapping, varIndexMapping, glyphOrder)
+            if getattr(vhvar, tableFields.sb1):
+                _remapVarIdxMap(vhvar, tableFields.sb1, varIndexMapping, glyphOrder)
+            if getattr(vhvar, tableFields.sb2):
+                _remapVarIdxMap(vhvar, tableFields.sb2, varIndexMapping, glyphOrder)
+            if tableFields.vOrigMapping and getattr(vhvar, tableFields.vOrigMapping):
```

Comment:
If the first check is to verify the table is VVAR, can we directly use tableTag == 'VVAR' instead, just to make the code easier to read.